Documentation/check-build.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# This was ported from the makefile, it can probably be converted
+# to Python (see e.g. https://stackoverflow.com/a/47657926), that
+# way we might be able to call Sphinx only once and make it quicker
+
+source_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+build_dir="${source_dir}/_build"
+warnings="${build_dir}/warnings.txt"
+spelling="${build_dir}/spelling/output.txt"
+
+target="${1:-"html"}"
+
+cd "${source_dir}"
+
+build_with_spellchecker() {
+    sphinx-build -b spelling -d "${build_dir}/doctrees" . "${build_dir}/spelling" -q -E 2>&1 > "${warnings}"
+}
+
+build_with_linkchecker() {
+    sphinx-build -b linkcheck -d "${build_dir}/doctrees" . "${build_dir}/spelling" -q -E
+}
+
+filter_warnings() {
+    grep -v \
+        -e "tabs assets" \
+        -e "misspelled words" \
+        -e "RemovedInSphinx20Warning" \
+        "${build_dir}/warnings.txt"
+}
+
+has_spelling_errors() {
+    test "$(wc -l < "${spelling}")" -ne 0 
+}
+
+describe_spelling_errors() {
+    printf "\nPlease fix the following spelling mistakes:\n"
+    # show file paths relative to repo root
+    sed 's/^/* Documentation\//' "${spelling}"
+}
+
+hint_about_wordlist_update() {
+    new_words="$(sed -E 's/^([^:]+:){2} \((.*)\)/\2/g' "${spelling}" | sort -u | tr -d '\r\n' | sed "s/'/\\\\\\\\'/g")"
+    printf "\nIf the words are not misspelled, run:\n%s %s\n" \
+        "Documentation/update-spelling_wordlist.sh" "${new_words}"
+}
+
+echo "Validating documentation (syntax, spelling)..."
+if build_with_spellchecker ; then
```

Comment:
I *think* that previously if the spellchecker build failed (non-zero exit code), we would exit immediately. With this script, I suspect we will just skip this conditional block and proceed to attempt to build again without the spellchecker target, so you'll end up running sphinx twice in such cases?

Reply:
Yes, that was intentional. What I was hoping is that we can catch our fatal warnings along with spelling errors. There ought to be a way to catch everything at once without running `sphinx-build` twice, but it seems like that would require native Python code (essentially extending `conf.py`), which would eliminate the need for grepping thing out etc, but I was hoping to avoid that for now.